Source/WebCore/Modules/mediasource/SourceBuffer.cpp

```diff
@@ -492,6 +493,10 @@ ExceptionOr<void> SourceBuffer::appendBufferInternal(const unsigned char* data,
     if (isRemoved() || m_updating)
         return Exception { InvalidStateError };
 
+    StringPrintStream message;
+    message.printf("SourceBuffer::appendBufferInternal(%p) - append size = %u, buffered = %s\n", this, size, toString(m_private->buffered()->ranges()).utf8().data());
+    DEBUG_LOG(LOGIDENTIFIER, message.toString());
```

Comment:
Doesn't DEBUG_LOG take printf-like parameters?

Reply:
No. It takes a list of parameters that are converted to String and joined. The problem with that is that it doesn't know how to convert all the types. Specifically, it doesn't know how to convert the `(SourceBuffer*)this` (or any other pointer).
I wish there was something like a `String::format()` method that generated a String from a printf-like template, but I haven't found it. The closest I found was `makeString()`, which uses the same conversion and joining strategy that DEBUG_LOG() uses, and has the same problems.